crates/fuel-core/src/database/sealed_block.rs

```diff
@@ -81,13 +81,12 @@ impl OnChainIterableKeyValueView {
     pub fn get_sealed_block_headers(
         &self,
         block_height_range: Range<u32>,
-    ) -> StorageResult<Vec<SealedBlockHeader>> {
+    ) -> StorageResult<Option<Vec<SealedBlockHeader>>> {
```

Comment:
yes, this is an intended change, because the function below this, `get_transactions_on_blocks` returns `StorageResult<Option<Vec<Transactions>>>`, was trying to make it similar in the return type

Reply:
it also makes it easier when dealing with the closures that we pass into `handle_request`

Reply:
Right, I think the last `.into_iter().collect()` bit here is redundant then.